### PR TITLE
Use anchors to avoid cache key duplication

### PR DESCRIPTION
**Pull request by f-meloni**
## Summary

Unify the different cache keys in one unique place to avoid duplication, and possible issues with typos.

## Test Plan

Check Circle CI jobs to verify that caches are working correctly

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -30,6 +30,22 @@ references:
     attach_workspace:
       at: *hermes_workspace_root
 
+  # -------------------------
+  #        Cache Key Anchors
+  # -------------------------
+  # Anchors for the cache keys
+
+  cache_keys:
+    gems_cache_key: &gems_cache_key v1-gems-{{ checksum "Gemfile.lock" }}
```
Comment (cipolleschi) [inline .circleci/config.yml:39]: I see that the `gem cache` key was [previously](https://github.com/facebook/react-native/pull/33821/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L99) only `1-`, now I see that it is `v1-`. Was it inteded or the force of habit?

**Review by cipolleschi — CHANGES_REQUESTED** [review]
Very good job! :clap: 
Thanks for the great job.

I left one question which I think was just an oversight.

Another nit change could be to sort the cache keys alphabetically. I know that it does not change anything practically, but it keeps the config more tidy. What do you think?

File: .circleci/config.yml (hunk 1 above)
Comment (f-meloni) [reply .circleci/config.yml:39]: Sorry, I forgot to add a comment here, this is the only one that was missing the v, so I've added it here as well :)

Comment (f-meloni) [reply .circleci/config.yml:41]: Correct, but from that moment on it will always find the key

**Review by cortinico — APPROVED** [review]

**Pull request closed without merging**